packages/graph/hash_graph/lib/graph/src/api/rest/entity.rs

```diff
@@ -340,5 +387,42 @@ async fn update_entity<P: StorePool + Send>(
     pool: Extension<Arc<P>>,
     body: Json<UpdateEntityRequest>,
 ) -> Result<Json<EntityMetadata>, StatusCode> {
-    todo!("Support updating of entities: https://app.asana.com/0/0/1203464975244303/f")
+    let Json(UpdateEntityRequest {
+        properties,
+        entity_id,
+        entity_type_id,
+        actor_id,
+        order,
+        archived,
+    }) = body;
+
+    let mut store = pool.acquire().await.map_err(|report| {
+        tracing::error!(error=?report, "Could not acquire store");
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?;
+
+    store
+        .update_entity(
+            entity_id,
+            None,
+            actor_id,
+            archived,
+            entity_type_id,
+            properties,
+            order,
+        )
+        .await
+        .map_err(|report| {
+            tracing::error!(error=?report, "Could not update entity");
+
+            if report.contains::<EntityDoesNotExist>() {
+                StatusCode::NOT_FOUND
+            } else if report.contains::<RaceConditionOnUpdate>() {
+                StatusCode::LOCKED
```

Comment:
It's fair enough we're using this return code, but I believe it's for WebDAV calendars specifically :D

Reply:
Happy to change it to another code. Just from the name it sounds reasonable to use it 😄

Reply:
We wouldn't have any clients that would interpret this as WebDAV i believe, but you should add it to the utoipa return code list in the proc macro :)

Reply:
Good point: 2c10386786917a2d635cd04829f0db2a196996a3